### PR TITLE
v0.2.0

### PR DESCRIPTION
**Pull request by scotow**
- Fix socket write
- Add sub crates re-exports
- Fix doc-tests & macros
- Feature gate yaml parsing
- Other minor changes

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -33,11 +36,11 @@ pub struct ConfigBuilder {
     use_tls: bool,
     async_buffer_size: Option<usize>,
     buffer_size: Option<usize>,
-    buffer_duration: Option<u64>,
+    buffer_duration: Option<Duration>,
     additional_fields: BTreeMap<Value, Value>,
     full_buffer_policy: Option<FullBufferPolicy>,
-    connect_timeout_ms: Option<u64>,
-    write_timeout_ms: Option<u64>,
+    connect_timeout_ms: Option<Duration>,
```
Comment (guillax) [inline src/config.rs]:
```suggestion
    connect_timeout: Option<Duration>,
```
I don't think the unit suffix is required anymore

File: src/config.rs [hunk 2]
```diff
@@ -33,11 +36,11 @@ pub struct ConfigBuilder {
...
+    write_timeout_ms: Option<Duration>,
```
Comment (guillax) [inline src/config.rs]:
```suggestion
    write_timeout: Option<Duration>,
```
Same here

File: src/config.rs [hunk 3]
```diff
@@ -157,22 +170,24 @@ impl ConfigBuilder {
...
-    /// If not set or set to `None`, `FullBufferPolicy::Discard` will be used by default
+    /// If not set or set to `None`, `FullBufferPolicy::Discard` will be used by
+    /// default
     pub fn set_full_buffer_policy(mut self, policy: Option<FullBufferPolicy>) -> ConfigBuilder {
         self.full_buffer_policy = policy;
         self
     }
     /// Set the TCP connect timeout.    
-    pub fn set_connect_timeout_ms(mut self, connect_timeout_ms: Option<u64>) -> ConfigBuilder {
+    pub fn set_connect_timeout_ms(mut self, connect_timeout_ms: Option<Duration>) -> ConfigBuilder {
```
Comment (guillax) [inline src/config.rs]:
```suggestion
    pub fn set_connect_timeout(mut self, connect_timeout: Option<Duration>) -> ConfigBuilder {
```
Let's remove all `_ms` suffixes where the duration was moved from `u64` to `std::time::Duration`.

File: src/config.rs [hunk 4]
```diff
@@ -157,22 +170,24 @@ impl ConfigBuilder {
...
         self.connect_timeout_ms = connect_timeout_ms;
         self
     }
     /// Set the TCP write timeout.    
-    pub fn set_write_timeout_ms(mut self, write_timeout_ms: Option<u64>) -> ConfigBuilder {
+    pub fn set_write_timeout_ms(mut self, write_timeout_ms: Option<Duration>) -> ConfigBuilder {
```
Comment (guillax) [inline src/config.rs]:
```suggestion
    pub fn set_write_timeout(mut self, write_timeout_ms: Option<Duration>) -> ConfigBuilder {
```
Let's remove all `_ms` suffixes where the duration was moved from `u64` to `std::time::Duration`.

File: src/config.rs [hunk 5]
```diff
@@ -225,22 +239,24 @@ pub struct Config {
     use_tls: bool,
     async_buffer_size: Option<usize>,
     buffer_size: Option<usize>,
-    buffer_duration: Option<u64>,
+    buffer_duration: Option<Duration>,
     additional_fields: BTreeMap<Value, Value>,
     full_buffer_policy: Option<FullBufferPolicy>,
-    connect_timeout_ms: Option<u64>,
-    write_timeout_ms: Option<u64>,
+    connect_timeout_ms: Option<Duration>,
```
Comment (guillax) [inline src/config.rs]:
```suggestion
    connect_timeout: Option<Duration>,
```

File: src/config.rs [hunk 6]
```diff
@@ -225,22 +239,24 @@ pub struct Config {
...
+    write_timeout_ms: Option<Duration>,
```
Comment (guillax) [inline src/config.rs]:
```suggestion
    write_timeout: Option<Duration>,
```

File: src/config.rs [hunk 7]
```diff
@@ -272,60 +288,72 @@ impl Config {
...
+
     /// Get the full buffer policy
     pub fn full_buffer_policy(&self) -> Option<FullBufferPolicy> {
         self.full_buffer_policy
     }
-    /// Get the write timeout in milliseconds
-    pub fn write_timeout_ms(&self) -> Option<u64> {
+
+    /// Get the write timeout
+    pub fn write_timeout_ms(&self) -> Option<Duration> {
```
Comment (guillax) [inline src/config.rs]:
```suggestion
    pub fn write_timeout(&self) -> Option<Duration> {
```

File: src/config.rs [hunk 8]
```diff
@@ -272,60 +288,72 @@ impl Config {
...
         self.write_timeout_ms
     }
-    /// Get the connect timeout in milliseconds
-    pub fn connect_timeout_ms(&self) -> Option<u64> {
+
+    /// Get the connect timeout
+    pub fn connect_timeout_ms(&self) -> Option<Duration> {
```
Comment (guillax) [inline src/config.rs]:
```suggestion
    pub fn connect_timeout(&self) -> Option<Duration> {
```

**Review by guillax — COMMENTED** [review]
Duration variables that have been moved from `u64` to `std::time::Duration` should not have kept their `_ms` suffix

**Comment by scotow** [comment]
`_ms` suffixes removed. @guillax

**Review by guillax — APPROVED** [review]

**Pull request merged**